File: 2493-reverse-odd-levels-of-binary-tree/reverse-odd-levels-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode reverseOddLevels(TreeNode root) {
        traverse(root.left, root.right, 0);
        return root;
    }

    static void traverse(TreeNode left, TreeNode right, int lvl){
        if(left == null || right == null) return;

        if(lvl%2 == 0){
            int temp = left.val;
            left.val = right.val;
            right.val = temp;
        }

        traverse(left.left, right.right, lvl+1);
        traverse(left.right, right.left, lvl+1);
    }
}

File: 2493-reverse-odd-levels-of-binary-tree/reverse-odd-levels-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    void traverse(TreeNode* leftNode, TreeNode* rightNode, int lvl) {
        
        if (!leftNode || !rightNode) {
            return;
        }

        
        if (lvl % 2 == 0) {
            std::swap(leftNode->val, rightNode->val);
        }

        traverse(leftNode->left, rightNode->right, lvl + 1);
        traverse(leftNode->right, rightNode->left, lvl + 1);
    }

    TreeNode* reverseOddLevels(TreeNode* root) {
        if (root) {
            traverse(root->left, root->right, 0);
        }
        return root;
    }
};